keps/sig-autoscaling/2021-scale-from-zero/README.md

```diff
@@ -0,0 +1,952 @@
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+With the addition of scaling based on object and external metrics it became possible to automatically adjust the number of running replicas based on an application
+provided metric. A typical use-case case for this is scaling the number of queue consumers based on the length of the consumed queue.
+
+In cases of a frequently idle queue or a less latency-sensitive workload, there is no need to keep one replica running at all times.
+Instead, you can dynamically scale to zero replicas, especially for workloads with high resource demands, such as those requiring GPUs.
+This approach not only reduces costs but also has significant energy-saving potential, particularly as GPU workloads become more prevalent.
+When replicas are scaled to zero, the HPA must also be capable of scaling back up as soon as messages become available.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Provide scaling to zero replicas for object and external metrics
+- Provide scaling from zero replicas for object and external metrics
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Provide scaling to/from zero replicas for resource metrics
+- Provide request buffering at the Kubernetes Service level
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Allow the HPA to scale from and to zero using `minReplicas: 0` when explicitly enabled with a flag.
```

Comment:
Nit: what is "with a flag"? Is that left over from the previous iteration before the condition was used to mark the scaled-to-zero? Or is that the feature gate?